nasa/earthdata-search
# EDSC-4105: As a user, I need to be able to submit an order through Earthdata Search Client that is over 2000 granules in size, and have all resulting requests be successfully submitted to Harmony for processing

# Overview

### What is the feature?

The issue was that for the particular order on the ticket as well as some subset of orders our memory allocation for the lambda is not sufficient. There is also an issue that the particular collection exposes where the order status was not updating in line with Harmony in cases where the state was `complete_with_errors`.

### What is the Solution?

Increased the lambda memory allocation to the next highest class; updated the logic in the `orderStatusItem` to account for this potential state coming back from Harmony

### What areas of the application does this impact?

Outgoing orders through Harmony via the `submitHarmonyOrder` lambda as well as the order state on the downloads page when the order is a Harmony order

# Testing

### Reproduction steps

- **Environment for testing:**
- **Collection to test with:**

1. Create this order on Harmony using these parameters `[/search?p=C1256358217-NSIDC_CUAT!C1256358217-NSIDC_CUAT&pg[1][v]=t&pg[1][m]=harmony0&pg[1][cd]=f&pg[1][ets]=t&pg[1][ess]=t&ee=uat&q=C1256358217-NSIDC_CUAT&sb[0]=-112.90495%2C58.01581%2C-11.09534%2C74.58281&tl=1722623855!3!!](https://search.sit.earthdata.nasa.gov/search?p=C1256358217-NSIDC_CUAT!C1256358217-NSIDC_CUAT&pg[1][v]=t&pg[1][m]=harmony0&pg[1][cd]=f&pg[1][ets]=t&pg[1][ess]=t&ee=uat&q=C1256358217-NSIDC_CUAT&sb[0]=-112.90495%2C58.01581%2C-11.09534%2C74.58281&tl=1722623855!3!!)`

2. Ensure that the order updates alongside of the orders on the Harmony workflow page `https://harmony.uat.earthdata.nasa.gov/workflow-ui` collection in in (UAT) and this ticket was for that

3. Ensure that when the order(s) complete the fraction updates and we get the Harmony message

### Attachments

<img width="1711" alt="image" src="https://github.com/user-attachments/assets/d26e278e-6a4c-461a-b0fb-3867257b8c88">

<img width="1684" alt="image" src="https://github.com/user-attachments/assets/0fa083e5-2a8c-43c4-85f8-8f44e553db11">


# Checklist
